packages/loader/container-loader/src/audience.ts

```diff
@@ -21,8 +21,12 @@ export class Audience extends EventEmitter implements IAudienceOwner {
      * Adds a new client to the audience
      */
     public addMember(clientId: string, details: IClient) {
-        this.members.set(clientId, details);
-        this.emit("addMember", clientId, details);
+        // Given that signal delivery is unreliable process, we might observe same client being added twice
+        // In such case we should see exactly same payload (IClient), and should not raise event twice!
```

Comment:
I'm a bit afraid to add such validation :)
@GaryWilber, is this something we can assert? The most likely case we run into this - IConnected.initialClients & early signals to container same client. It might be self, or some other clients that happened to join at the same time.
Is it guaranteed that IClient info in both to be exactly the same?

@andre4i - I'll consider adding it as separate PR, as I have a feeling it may backfire, and I might need to back it out or forward fix.

Reply:
I can speak for the latest logic, where deli is in charge of the join/leave signals. 
Clients are added to `initialClients` from [this code](https://github.com/microsoft/FluidFramework/blob/d9fd91378504eb2d352c7c1f090bf3cff8b62252/server/routerlicious/packages/lambdas/src/broadcaster/lambda.ts#L99). That is the broadcaster lambda. It will broadcast the signal after inserting it into `initialClients`

So the `IClient` object in join/leave signals will be the same `IClient` as the one in `initialClients` because both are sourced from `ticketedSignalMessage.operation`

Reply:
Ok, thanks. Opened https://dev.azure.com/fluidframework/internal/_workitems/edit/2033 to follow up with asserts